content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,361 +35,643 @@ are covered here as well as the `System.Diagnostics` API.
+    using (var myActivity = activitySource.StartActivity("rollTheDice"))
+    {
+        for (int i = 0; i < rolls; i++)
+        {
+            results.Add(rollOnce());
+        }
 
-### Add tags to an Activity
+        return results;
+    }
+}
 
-Tags (the equivalent of
-[`Attributes`](/docs/concepts/signals/traces/#attributes) in OpenTelemetry) let
-you attach key/value pairs to an `Activity` so it carries more information about
-the current operation that it's tracking.
+```
+If you followed the instructions using the [example app](#example-app) up to
+this point, you can copy the code above in your library file `Dice.cs`. You
+should now be able to see activities/spans emitted from your app.
 
-```csharp
-using var myActivity = MyActivitySource.StartActivity("SayHello");
+Start your app as follows, and then send it requests by visiting <http://localhost:8080/rolldice?rolls=12> with your browser or curl.
 
-activity?.SetTag("operation.value", 1);
-activity?.SetTag("operation.name", "Saying hello!");
-activity?.SetTag("operation.other-stuff", new int[] { 1, 2, 3 });
+```sh
+dotnet run
 ```
 
-We recommend that all Tag names are defined in constants rather than defined
-inline as this provides both consistency and also discoverability.
+After a while, you should see the spans printed in the console by the
+`ConsoleExporter`, something like this:
+
+```json
+Activity.TraceId:            841d70616c883db82b4ae4e11c728636
+Activity.SpanId:             9edfe4d69b0d6d8b
+Activity.TraceFlags:         Recorded
+Activity.ParentSpanId:       39fcd105cf958377
+Activity.ActivitySourceName: dice-server
+Activity.DisplayName:        rollTheDice
+Activity.Kind:               Internal
+Activity.StartTime:          2024-04-10T15:24:00.3620354Z
+Activity.Duration:           00:00:00.0144329
+Resource associated with Activity:
+    service.name: dice-server
+    service.version: 1.0.0
+    service.instance.id: 7a7a134f-3178-4ac6-9625-96df77cff8b4
+    telemetry.sdk.name: opentelemetry
+    telemetry.sdk.language: dotnet
+    telemetry.sdk.version: 1.7.0
 
-### Adding events
+```
 
-An [event](/docs/concepts/signals/traces/#span-events) is a human-readable
-message on an `Activity` that represents "something happening" during its
-lifetime.
+### Create nested Activities
+Nested [spans](/docs/concepts/signals/traces/#spans) let you track work that's
+nested in nature. For example, the `rollOnce()` function below represents a
+nested operation. The following sample creates a nested span that tracks
+`rollOnce()`:
 
 ```csharp
-using var myActivity = MyActivitySource.StartActivity("SayHello");
+private int rollOnce()
+{
+    using (var childActivity = activitySource.StartActivity("rollOnce"))
+    {
+      int result;
+      
+      result = Random.Shared.Next(min, max + 1);
+        
+      return result;
+    }
+}
+
+```
 
-// ...
+When you view the spans in a trace visualization tool, `rollOnce` childActivity will be
```

Comment:
for demonstrating the idea of child spans, this is very good! I'd also want to keep adequate warning here for users to avoid too much activity, as that'll expensive, but more importantly - makes visualization harder.